odoo/http.py

```diff
@@ -1176,6 +1176,33 @@ def update_trace(self, request):
         self.is_dirty = True
         return new_trace
 
+    def get_last_ip(self):
+        """
+            :return str: the IP address of last trace in the session, if no trace returns `None`
+        """
+        try:
+            return self['_trace'][-1]['ip_address']
+        except IndexError:
+            return None
+
+    def has_history_ip(self):
+        """
+            :return bool: `True` if more than one trace in the session, `False` otherwise
+        """
+        return len(self['_trace']) > 1
+
+    def get_history_ips(self, number_trusted_days):
+        """
+            :param number_trusted_days: number of days during which ip addresses are known
+                                        since the last trace
+            :return: history of ip address from most recent to least recent trace
+        """
+        if not self.has_history_ip():
+            return
+        last_trace_activity = self['_trace'][-1]['last_activity']
+        time_ = last_trace_activity - int(timedelta(days=number_trusted_days).total_seconds())
+        for trace in filter(lambda t: t['last_activity'] >= time_, reversed(self['_trace'][:-1])):
+            yield trace['ip_address']
```

Comment:
Since these 3 methods are used only in one place, `_check_location`, I am not sure it is worth to factorize them.
+ by putting them here you make it kinda "official" that it can be used / it's part of the API, and you are less free to make changes if you need to change something, to not break the stable policy.